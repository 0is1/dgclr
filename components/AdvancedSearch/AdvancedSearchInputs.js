// @flow
import React, { Component } from 'react';
import { connect } from 'react-redux';
import {
  Box, Flex, Label, Text,
} from 'rebass';
import update from 'updeep';
import { omit, size } from 'lodash';

import RatingSelect from 'components/Select/RatingSelect';
import BasketTypeSelect from 'components/Select/BasketTypeSelect';
import TeeTypeSelect from 'components/Select/TeeTypeSelect';
import SurfaceTypeSelect from 'components/Select/SurfaceTypeSelect';
import CourseTypeSelect from 'components/Select/CourseTypeSelect';
import AdvancedSearchMap from 'components/Map/AdvancedSearchMap';
import Toggle from 'components/Toggle';
import colors from 'components/colors';
import Input from 'components/Input';
import type { CoordinatesObject, State } from 'lib/types';
import { convertMetersToKilometers } from 'helpers/utils';
import { ADVANCED_NEARBY, ADVANCED_COURSE_INFO } from 'lib/constants';
import { INPUT_FILTER_NAMES } from 'components/Input/constants';
import { SELECT_FILTER_NAMES } from 'components/Select/constants';
import RebassComponents from 'components/RebassComponents';
import {
  getCurrentAdvancedFilter,
  isAdvancedSearchMapVisible,
  isAllAdvancedSearchInputsOpen,
} from './selectors';
import {
  setCurrentAdvancedSearchFilter,
  toggleAdvancedSearchMap,
  toggleAdvancedSearchInputs as toggleAdvancedSearchInputsFunc,
} from './actions';
import Styles from './AdvancedSearchInputs.styles';

const { Divider } = RebassComponents;
const { FadeInBox } = Styles;

type Props = {};
type MapStateToProps = {
  filter: string,
  mapChecked: boolean,
  allInputsOpen: boolean,
};

type MapDispatchToProps = {
  setAdvancedSearchFilter: Function,
  toggleMapVisibility: Function,
  toggleAdvancedSearchInputs: Function,
};

type CombinedProps = Props & MapStateToProps & MapDispatchToProps;
type FilterType = { courseInfo: {} };

export class AdvancedSearchInputs extends Component<CombinedProps> {
  getParsedFilter = () => {
    const { filter } = this.props;
    return !filter.length ? {} : JSON.parse(filter);
  };

  setAdvancedSearchFilterData = (newFilter: FilterType) => {
    const { setAdvancedSearchFilter } = this.props;
    // console.log('newFilter: ', this.cleanCourseInfo(newFilter));
    setAdvancedSearchFilter(JSON.stringify(this.cleanCourseInfo(newFilter)));
  };

  cleanCourseInfo = (newFilter: FilterType) => {
    if (newFilter.courseInfo && size(newFilter.courseInfo) === 0) return omit(newFilter, [ADVANCED_COURSE_INFO]);
    return newFilter;
  };

  onRatingChange = (rating: []) => {
    // console.log('onRatingChange: ', rating);
    const filter = this.getParsedFilter();
    // console.log('onRatingChange filter: ', filterData);
    const newFilter = rating.length > 0
      ? update({ rating }, filter)
      : omit(filter, [SELECT_FILTER_NAMES.rating.filterName]);
    this.setAdvancedSearchFilterData(newFilter);
  };

  onBasketTypeChange = (basketType: string) => {
    const filter = this.getParsedFilter();
    // console.log('onBasketTypeChange filter: ', filter);
    const newFilter = basketType.length > 0
      ? update({ courseInfo: { basketType } }, filter)
      : {
        ...filter,
        courseInfo: omit(filter.courseInfo, [
          SELECT_FILTER_NAMES.basketType.filterName,
        ]),
      };
    this.setAdvancedSearchFilterData(newFilter);
  };

  onTeeTypeChange = (teeType: string) => {
    const filter = this.getParsedFilter();
    // console.log('onTeeTypeChange filter: ', filter);
    const newFilter = teeType.length > 0
      ? update({ courseInfo: { teeType } }, filter)
      : {
        ...filter,
        courseInfo: omit(filter.courseInfo, [
          SELECT_FILTER_NAMES.teeType.filterName,
        ]),
      };
    this.setAdvancedSearchFilterData(newFilter);
  };

  onSurfaceTypeChange = (surfaceShapeTypes: string) => {
    const filter = this.getParsedFilter();
    // console.log('surfaceType filter: ', filter);
    const newFilter = surfaceShapeTypes.length > 0
      ? update({ courseInfo: { surfaceShapeTypes } }, filter)
      : {
        ...filter,
        courseInfo: omit(filter.courseInfo, [
          SELECT_FILTER_NAMES.surfaceShapeTypes.filterName,
        ]),
      };
    this.setAdvancedSearchFilterData(newFilter);
  };

  onCourseTypeChange = (courseTypes: string) => {
    const filter = this.getParsedFilter();
    // console.log('courseTypes filter: ', filter);
    const newFilter = courseTypes.length > 0
      ? update({ courseInfo: { courseTypes } }, filter)
      : {
        ...filter,
        courseInfo: omit(filter.courseInfo, [
          SELECT_FILTER_NAMES.courseTypes.filterName,
        ]),
      };
    this.setAdvancedSearchFilterData(newFilter);
  };

  onMapSearchChange = (data: {
    coordinates: CoordinatesObject,
    radius: number,
  }) => {
    const filter = this.getParsedFilter();
    //  list the longitude first and then latitude https://docs.mongodb.com/manual/reference/geojson/#geojson-point
    const nearby = data.coordinates
      && data.radius && {
      maxDistance: convertMetersToKilometers(parseInt(data.radius, 10)),
      coordinates: [data.coordinates.lng, data.coordinates.lat],
    };
    if (nearby) {
      this.setAdvancedSearchFilterData(update({ nearby }, filter));
      return;
    }
    this.omitMapFilter();
  };

  omitMapFilter = () => {
    const filter = this.getParsedFilter();
    this.setAdvancedSearchFilterData(omit(filter, [ADVANCED_NEARBY]));
  };

  handleMapToggle = (mapChecked: boolean) => {
    if (!mapChecked) this.omitMapFilter();
    const { toggleMapVisibility } = this.props;
    toggleMapVisibility(mapChecked);
  };

  toggleAdvancedSearchInputs = () => {
    const { allInputsOpen, toggleAdvancedSearchInputs } = this.props;
    toggleAdvancedSearchInputs(!allInputsOpen);
  };

  onHoleAverageLengthChange = (values: Array<number>) => {
    const filter = this.getParsedFilter();
    const [min, max] = values;
    const newFilter = update(
      { [INPUT_FILTER_NAMES.holeAverageLength.filterName]: { min, max } },
      filter,
    );
    this.setAdvancedSearchFilterData(newFilter);
  };

  render() {
    const { allInputsOpen, mapChecked } = this.props;
    return (
      <Box>
        <Text fontWeight="700" my={2} width="100%">
          Edistynyt haku (varhaisessa kehitysvaiheessa)
        </Text>
        <Flex flexWrap="wrap">
          <Box
            pr={[0, 0, '.5rem', '.5rem']}
            mb=".75rem"
            width={[1, 1, 1 / 2, 1 / 2]}
          >
            <RatingSelect onChange={this.onRatingChange} />
          </Box>
          <Box
            pr={[0, 0, '.5rem', '.5rem']}
            mb=".75rem"
            width={[1, 1, 1 / 2, 1 / 2]}
          >
            <CourseTypeSelect onChange={this.onCourseTypeChange} />
          </Box>
          <Box
            pr={[0, 0, '.5rem', '.5rem']}
            mb=".75rem"
            width={[1, 1, 1 / 2, 1 / 2]}
          >
            <Label>Väylien keskipituus:</Label>
            <Input
              onChange={this.onHoleAverageLengthChange}
              options={{
                type: 'slider',
                initialValues: [40, 180],
                domain: [10, 180],
                step: 10,
                filterName: INPUT_FILTER_NAMES.holeAverageLength.filterName,
              }}
            />
          </Box>
          <FadeInBox show={allInputsOpen}>
            {allInputsOpen && (
              <>
                <Box
                  pr={[0, 0, '.5rem', '.5rem']}
                  mb=".75rem"
                  width={[1, 1, 1 / 2, 1 / 2]}
                >
                  <BasketTypeSelect onChange={this.onBasketTypeChange} />
                </Box>
                <Box
                  pr={[0, 0, '.5rem', '.5rem']}
                  mb=".75rem"
                  width={[1, 1, 1 / 2, 1 / 2]}
                >
                  <TeeTypeSelect onChange={this.onTeeTypeChange} />
                </Box>
                <Box
                  pr={[0, 0, '.5rem', '.5rem']}
                  mb=".75rem"
                  width={[1, 1, 1 / 2, 1 / 2]}
                >
                  <SurfaceTypeSelect onChange={this.onSurfaceTypeChange} />
                </Box>
              </>
            )}
          </FadeInBox>
          <Toggle
            label="Näytä kaikki valinnat:"
            checked={allInputsOpen}
            handleOnChange={this.toggleAdvancedSearchInputs}
          />
          <Box mb=".75rem" width={[1]}>
            <Toggle
              label="Käytä karttahakua:"
              checked={mapChecked}
              handleOnChange={this.handleMapToggle}
            />
            <AdvancedSearchMap
              mapVisible={mapChecked}
              handleChange={this.onMapSearchChange}
            />
          </Box>
        </Flex>
        <Divider w={1} borderColor={colors.info} />
      </Box>
    );
  }
}

const mapStateToProps = (state: State): MapStateToProps => ({
  allInputsOpen: isAllAdvancedSearchInputsOpen(state),
  filter: getCurrentAdvancedFilter(state),
  mapChecked: isAdvancedSearchMapVisible(state),
});
const mapDispatchToProps = (dispatch: Function): MapDispatchToProps => ({
  setAdvancedSearchFilter: filter => dispatch(setCurrentAdvancedSearchFilter(filter)),
  toggleMapVisibility: (visible: boolean) => dispatch(toggleAdvancedSearchMap(visible)),
  toggleAdvancedSearchInputs: (visible: boolean) => dispatch(toggleAdvancedSearchInputsFunc(visible)),
});

export default connect<CombinedProps, Props, any, any, any, Function>(
  mapStateToProps,
  mapDispatchToProps,
)(AdvancedSearchInputs);
