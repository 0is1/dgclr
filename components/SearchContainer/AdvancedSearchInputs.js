// @flow
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Box, Flex, Text } from 'rebass';
import update from 'updeep';
import { omit, size } from 'lodash';
import { getCurrentAdvancedFilter } from 'components/SearchContainer/selectors';
import { setCurrentAdvancedSearchFilter } from 'components/SearchContainer/actions';
import RatingSelect from 'components/Select/RatingSelect';
import BasketTypeSelect from 'components/Select/BasketTypeSelect';
import TeeTypeSelect from 'components/Select/TeeTypeSelect';
import SurfaceTypeSelect from 'components/Select/SurfaceTypeSelect';
import AdvancedSearchMap from 'components/Map/AdvancedSearchMap';
import colors from 'components/colors';
import type { CoordinatesObject, State } from 'lib/types';
import { convertMetersToKilometers } from 'helpers/utils';
import {
  ADVANCED_RATING,
  ADVANCED_NEARBY,
  ADVANCED_TEE_TYPE,
  ADVANCED_BASKET_TYPE,
  ADVANCED_COURSE_INFO,
  ADVANCED_SURFACE_SHAPE_TYPES,
} from 'lib/constants';
import RebassComponents from 'components/RebassComponents';

const { Divider } = RebassComponents;
type Props = { filter: string, setFilter: Function };

class AdvancedSearchInputs extends Component<Props> {
  getParsedFilter = () => {
    const { filter } = this.props;
    return !filter.length ? {} : JSON.parse(filter);
  };

  setFilterData = (newFilter) => {
    const { setFilter } = this.props;
    // console.log('newFilter: ', this.cleanCourseInfo(newFilter));
    setFilter(JSON.stringify(this.cleanCourseInfo(newFilter)));
  };

  cleanCourseInfo = (newFilter) => {
    if (newFilter.courseInfo && size(newFilter.courseInfo) === 0) return omit(newFilter, [ADVANCED_COURSE_INFO]);
    return newFilter;
  };

  onRatingChange = (rating: []) => {
    // console.log('onRatingChange: ', rating);
    const filter = this.getParsedFilter();
    // console.log('onRatingChange filter: ', filterData);
    const newFilter = rating.length > 0 ? update({ rating }, filter) : omit(filter, [ADVANCED_RATING]);
    this.setFilterData(newFilter);
  };

  onBasketTypeChange = (basketType: string) => {
    const filter = this.getParsedFilter();
    // console.log('onBasketTypeChange filter: ', filter);
    const newFilter = basketType.length > 0
      ? update({ courseInfo: { basketType } }, filter)
      : { ...filter, courseInfo: omit(filter.courseInfo, [ADVANCED_BASKET_TYPE]) };
    this.setFilterData(newFilter);
  };

  onTeeTypeChange = (teeType: string) => {
    const filter = this.getParsedFilter();
    // console.log('onTeeTypeChange filter: ', filter);
    const newFilter = teeType.length > 0
      ? update({ courseInfo: { teeType } }, filter)
      : { ...filter, courseInfo: omit(filter.courseInfo, [ADVANCED_TEE_TYPE]) };
    this.setFilterData(newFilter);
  };

  onSurfaceTypeChange = (surfaceShapeTypes: string) => {
    const filter = this.getParsedFilter();
    // console.log('surfaceType filter: ', filter);
    const newFilter = surfaceShapeTypes.length > 0
      ? update({ courseInfo: { surfaceShapeTypes } }, filter)
      : { ...filter, courseInfo: omit(filter.courseInfo, [ADVANCED_SURFACE_SHAPE_TYPES]) };
    this.setFilterData(newFilter);
  };

  onMapSearchChange = (data: { coordinates: CoordinatesObject, radius: number }) => {
    const filter = this.getParsedFilter();
    const nearby = data.coordinates
      && data.radius && {
      maxDistance: convertMetersToKilometers(parseInt(data.radius, 10)),
      coordinates: [data.coordinates.lat, data.coordinates.lng],
    };
    const newFilter = nearby ? update({ nearby }, filter) : { ...filter, nearby: omit(filter, [ADVANCED_NEARBY]) };
    this.setFilterData(newFilter);
  };

  render() {
    return (
      <Box>
        <Text fontWeight="700" my={2} width="100%">
          Edistynyt haku (eritt√§in varhaisessa kehitysvaiheessa)
        </Text>
        <Flex flexWrap="wrap">
          <Box pr={[0, 0, '.5rem', '.5rem']} mb=".75rem" width={[1, 1, 1 / 2, 1 / 2]}>
            <RatingSelect onChange={this.onRatingChange} />
          </Box>
          <Box pl={[0, 0, '.5rem', '.5rem']} mb=".75rem" width={[1, 1, 1 / 2, 1 / 2]}>
            <BasketTypeSelect onChange={this.onBasketTypeChange} />
          </Box>
          <Box pr={[0, 0, '.5rem', '.5rem']} mb=".75rem" width={[1, 1, 1 / 2, 1 / 2]}>
            <TeeTypeSelect onChange={this.onTeeTypeChange} />
          </Box>
          <Box pl={[0, 0, '.5rem', '.5rem']} mb=".75rem" width={[1, 1, 1 / 2, 1 / 2]}>
            <SurfaceTypeSelect onChange={this.onSurfaceTypeChange} />
          </Box>
          <Box pl={[0, 0, '.5rem', '.5rem']} mb=".75rem" width={[1]}>
            <AdvancedSearchMap onChange={this.onMapSearchChange} />
          </Box>
        </Flex>
        <Divider w={1} borderColor={colors.info} />
      </Box>
    );
  }
}

const mapStateToProps = (state: State) => ({
  filter: getCurrentAdvancedFilter(state),
});
const mapDispatchToProps = dispatch => ({
  setFilter: filter => dispatch(setCurrentAdvancedSearchFilter(filter)),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(AdvancedSearchInputs);
