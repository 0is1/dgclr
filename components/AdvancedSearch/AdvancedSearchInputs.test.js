import React from 'react';
import { shallow } from 'enzyme';
import { AdvancedSearchInputs } from './AdvancedSearchInputs';

const mockProps = (props = {}) => ({
  filter: '{}',
  mapChecked: true,
  allInputsOpen: false,
  ...props,
});

describe('<AdvancedSearchInputs />', () => {
  describe('snapshots', () => {
    it('renders with default props', () => {
      const props = mockProps();
      const subject = shallow(<AdvancedSearchInputs {...props} />);
      expect(subject).toMatchSnapshot();
    });
    it('renders with allInputsOpen', () => {
      const props = mockProps({ allInputsOpen: true });
      const subject = shallow(<AdvancedSearchInputs {...props} />);
      expect(subject).toMatchSnapshot();
    });
  });
});
