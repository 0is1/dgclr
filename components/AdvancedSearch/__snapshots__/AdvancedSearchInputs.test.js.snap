// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AdvancedSearchInputs /> snapshots renders with allInputsOpen 1`] = `
<Box>
  <Text
    fontWeight="700"
    m={0}
    my={2}
    width="100%"
  >
    Edistynyt haku (varhaisessa kehitysvaiheessa)
  </Text>
  <Flex
    extend={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "_foldedDefaultProps": Object {},
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": false,
          "rules": Array [
            "box-sizing: border-box;",
            [Function],
            [Function],
            [Function],
            [Function],
            [Function],
            [Function],
            [Function],
            [Function],
          ],
        },
        "displayName": "Box",
        "foldedComponentIds": Array [],
        "propTypes": Object {
          "alignSelf": [Function],
          "bg": [Function],
          "color": [Function],
          "flex": [Function],
          "fontSize": [Function],
          "m": [Function],
          "mb": [Function],
          "ml": [Function],
          "mr": [Function],
          "mt": [Function],
          "mx": [Function],
          "my": [Function],
          "order": [Function],
          "p": [Function],
          "pb": [Function],
          "pl": [Function],
          "pr": [Function],
          "pt": [Function],
          "px": [Function],
          "py": [Function],
          "width": [Function],
        },
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "systemComponent": true,
        "target": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    flexWrap="wrap"
  >
    <Box
      mb=".75rem"
      pr={
        Array [
          0,
          0,
          ".5rem",
          ".5rem",
        ]
      }
      width={
        Array [
          1,
          1,
          0.5,
          0.5,
        ]
      }
    >
      <Connect(RatingSelect)
        onChange={[Function]}
      />
    </Box>
    <Box
      mb=".75rem"
      pr={
        Array [
          0,
          0,
          ".5rem",
          ".5rem",
        ]
      }
      width={
        Array [
          1,
          1,
          0.5,
          0.5,
        ]
      }
    >
      <CourseTypeSelect
        onChange={[Function]}
      />
    </Box>
    <Box
      mb=".75rem"
      pr={
        Array [
          0,
          0,
          ".5rem",
          ".5rem",
        ]
      }
      width={
        Array [
          1,
          1,
          0.5,
          0.5,
        ]
      }
    >
      <Label
        alignItems="center"
        fontSize={1}
        is="label"
        mb={1}
      >
        Väylien keskipituus:
      </Label>
      <InputComponent
        focusOnMount={false}
        onChange={[Function]}
        options={
          Object {
            "domain": Array [
              10,
              180,
            ],
            "filterName": "holeAverageLength",
            "initialValues": Array [
              40,
              180,
            ],
            "step": 10,
            "type": "slider",
          }
        }
        placeholder=""
        value=""
      />
    </Box>
    <AdvancedSearchInputsstyles__FadeInBox
      show={true}
    >
      <Box
        mb=".75rem"
        pr={
          Array [
            0,
            0,
            ".5rem",
            ".5rem",
          ]
        }
        width={
          Array [
            1,
            1,
            0.5,
            0.5,
          ]
        }
      >
        <BasketTypeSelect
          onChange={[Function]}
        />
      </Box>
      <Box
        mb=".75rem"
        pr={
          Array [
            0,
            0,
            ".5rem",
            ".5rem",
          ]
        }
        width={
          Array [
            1,
            1,
            0.5,
            0.5,
          ]
        }
      >
        <TeeTypeSelect
          onChange={[Function]}
        />
      </Box>
      <Box
        mb=".75rem"
        pr={
          Array [
            0,
            0,
            ".5rem",
            ".5rem",
          ]
        }
        width={
          Array [
            1,
            1,
            0.5,
            0.5,
          ]
        }
      >
        <SurfaceTypeSelect
          onChange={[Function]}
        />
      </Box>
    </AdvancedSearchInputsstyles__FadeInBox>
    <Toggle
      checked={true}
      handleOnChange={[Function]}
      label="Näytä kaikki valinnat:"
    />
    <Box
      mb=".75rem"
      width={
        Array [
          1,
        ]
      }
    >
      <Toggle
        checked={true}
        handleOnChange={[Function]}
        label="Käytä karttahakua:"
      />
      <Connect(AdvancedSearchMap)
        handleChange={[Function]}
        mapVisible={true}
      />
    </Box>
  </Flex>
  <Dividerstyles
    borderColor="#6173fb"
    w={1}
  />
</Box>
`;

exports[`<AdvancedSearchInputs /> snapshots renders with default props 1`] = `
<Box>
  <Text
    fontWeight="700"
    m={0}
    my={2}
    width="100%"
  >
    Edistynyt haku (varhaisessa kehitysvaiheessa)
  </Text>
  <Flex
    extend={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "_foldedDefaultProps": Object {},
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": false,
          "rules": Array [
            "box-sizing: border-box;",
            [Function],
            [Function],
            [Function],
            [Function],
            [Function],
            [Function],
            [Function],
            [Function],
          ],
        },
        "displayName": "Box",
        "foldedComponentIds": Array [],
        "propTypes": Object {
          "alignSelf": [Function],
          "bg": [Function],
          "color": [Function],
          "flex": [Function],
          "fontSize": [Function],
          "m": [Function],
          "mb": [Function],
          "ml": [Function],
          "mr": [Function],
          "mt": [Function],
          "mx": [Function],
          "my": [Function],
          "order": [Function],
          "p": [Function],
          "pb": [Function],
          "pl": [Function],
          "pr": [Function],
          "pt": [Function],
          "px": [Function],
          "py": [Function],
          "width": [Function],
        },
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "systemComponent": true,
        "target": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    flexWrap="wrap"
  >
    <Box
      mb=".75rem"
      pr={
        Array [
          0,
          0,
          ".5rem",
          ".5rem",
        ]
      }
      width={
        Array [
          1,
          1,
          0.5,
          0.5,
        ]
      }
    >
      <Connect(RatingSelect)
        onChange={[Function]}
      />
    </Box>
    <Box
      mb=".75rem"
      pr={
        Array [
          0,
          0,
          ".5rem",
          ".5rem",
        ]
      }
      width={
        Array [
          1,
          1,
          0.5,
          0.5,
        ]
      }
    >
      <CourseTypeSelect
        onChange={[Function]}
      />
    </Box>
    <Box
      mb=".75rem"
      pr={
        Array [
          0,
          0,
          ".5rem",
          ".5rem",
        ]
      }
      width={
        Array [
          1,
          1,
          0.5,
          0.5,
        ]
      }
    >
      <Label
        alignItems="center"
        fontSize={1}
        is="label"
        mb={1}
      >
        Väylien keskipituus:
      </Label>
      <InputComponent
        focusOnMount={false}
        onChange={[Function]}
        options={
          Object {
            "domain": Array [
              10,
              180,
            ],
            "filterName": "holeAverageLength",
            "initialValues": Array [
              40,
              180,
            ],
            "step": 10,
            "type": "slider",
          }
        }
        placeholder=""
        value=""
      />
    </Box>
    <AdvancedSearchInputsstyles__FadeInBox
      show={false}
    />
    <Toggle
      checked={false}
      handleOnChange={[Function]}
      label="Näytä kaikki valinnat:"
    />
    <Box
      mb=".75rem"
      width={
        Array [
          1,
        ]
      }
    >
      <Toggle
        checked={true}
        handleOnChange={[Function]}
        label="Käytä karttahakua:"
      />
      <Connect(AdvancedSearchMap)
        handleChange={[Function]}
        mapVisible={true}
      />
    </Box>
  </Flex>
  <Dividerstyles
    borderColor="#6173fb"
    w={1}
  />
</Box>
`;
